packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -847,7 +857,19 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
   set selection(TextSelection value) {
     if (_selection == value)
       return;
-    _selection = value;
+    // We use the _fallbackAffinity when the set selection has a null
+    // affinity. This happense when the platform does not supply affinity,
+    // in which case using the fallback affinity computed from dart:ui will
+    // be superior to simply defaulting to TextAffinity.downstream.
+    if (value.affinity == null) {
+      _selection = TextSelection(
+        baseOffset: value.baseOffset,
+        extentOffset: value.extentOffset,
+        affinity: _fallbackAffinity,
+      );
```

Comment:
Would it be more robust to use `value.copyWith` here, or did you purposely avoid it?

Reply:
Ahh you are right, should just use this.